subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/SubscriptionServiceTests.kt

```diff
@@ -1062,6 +1102,60 @@ class SubscriptionServiceTests : WithTimescaleContainer, WithKafkaContainer {
             .shouldSucceedWith { assertEquals(0, it.size) }
     }
 
+    @Test
+    fun `it should not return a subscription if throttling has not elapsed yet`() = runTest {
+        val expandedEntity = expandJsonLdEntity(entity, APIC_COMPOUND_CONTEXTS)
+
+        val payload = mapOf(
+            "id" to "urn:ngsi-ld:Beehive:1234567890".toUri(),
+            "type" to NGSILD_SUBSCRIPTION_TERM,
+            "watchedAttributes" to listOf(INCOMING_COMPACT_PROPERTY),
+            "notification" to mapOf(
+                "endpoint" to mapOf("uri" to "http://my.endpoint/notifiy"),
+                "lastNotification" to ngsiLdDateTime()
+            ),
+            "throttling" to 300
+        )
+
+        val subscription = ParsingUtils.parseSubscription(payload, emptyList()).shouldSucceedAndResult()
+
+        subscriptionService.create(subscription, mockUserSub).shouldSucceed()
+
+        subscriptionService.getMatchingSubscriptions(
+            expandedEntity,
+            setOf(INCOMING_COMPACT_PROPERTY),
+            ATTRIBUTE_UPDATED
+        )
+            .shouldSucceedWith { assertEquals(0, it.size) }
+    }
+
+    @Test
+    fun `it should return a subscription if throttling has elapsed`() = runTest {
+        val expandedEntity = expandJsonLdEntity(entity, APIC_COMPOUND_CONTEXTS)
+
+        val payload = mapOf(
+            "id" to "urn:ngsi-ld:Beehive:1234567890".toUri(),
+            "type" to NGSILD_SUBSCRIPTION_TERM,
+            "watchedAttributes" to listOf(INCOMING_COMPACT_PROPERTY),
+            "notification" to mapOf(
+                "endpoint" to mapOf("uri" to "http://my.endpoint/notifiy"),
+                "lastNotification" to ngsiLdDateTime().minusMinutes(5)
+            ),
+            "throttling" to 200
+        )
+
+        val subscription = ParsingUtils.parseSubscription(payload, emptyList()).shouldSucceedAndResult()
+
+        subscriptionService.create(subscription, mockUserSub).shouldSucceed()
+
+        subscriptionService.getMatchingSubscriptions(
+            expandedEntity,
+            setOf(INCOMING_COMPACT_PROPERTY),
+            ATTRIBUTE_UPDATED
+        )
+            .shouldSucceedWith { assertEquals(0, it.size) }
```

Comment:
the purpose of this test is to check that you have one subscription when `throttling` has elapsed.

you can add a delay similar to what is done in https://github.com/stellio-hub/stellio-context-broker/blob/develop/subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/SubscriptionServiceTests.kt#L347-L350